src/popup/modules/UserInterface.js

```diff
@@ -435,6 +437,37 @@ function triggerFileSave(file, filename, requestDownloadPermissions) {
     });
 }
 
+/**
+ * If qrcode text was an URL i.e contains :// this function generates a filename using the domain and replacing all special characters with _
+ *.If not return "qrcode"
+ *
+ * @function
+ * @private
+ * @param { } 
+ * @returns {filename}
+ */
+function generateFilename(){
+	var firstIndex = qrCodeInputText.indexOf("://");
```

Comment:
Yeah `indexOf` is possible. Just to let you know: [JS has newer methods like `.includes`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String/includes) that are more intuitive to read and maybe using that here would be better?

Reply:
Or wait… [`startsWith`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) would be even better? Cause we don't want to return the domain only, if the input e.g. is `Check out the URL: https://test.com` the qr code would be named `test` or so, which would be bad IMHO.

So rather be more strict here…

Reply:
This works also for http:// and ftp:// so why change?

Reply:
I want to concentrate on the basics and 95% of the assumed usage and make it KISS.
The user still has the possibility to change the filename in the system download dialog.